src/js/components/deployments/deployment-wizard/phasesettings.js

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import pluralize from 'pluralize';
+
+import Chip from '@material-ui/core/Chip';
+import AddIcon from '@material-ui/icons/Add';
+
+import { Table, TableBody, TableCell, TableHead, TableRow, Select, MenuItem, Input, InputAdornment, IconButton } from '@material-ui/core';
+import CancelIcon from '@material-ui/icons/Cancel';
+
+export default class PhaseSettings extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+  }
+
+  getRemainder(phases) {
+    // use this to get remaining percent of last phase so we don't set a hard number
+    let remainder = 100;
+    for (let phase of phases) {
+      remainder = phase.batch_size ? remainder - phase.batch_size : remainder;
+    }
+    return remainder;
+  }
+
+  updateDelay(value, index) {
+    let newPhases = this.props.phases;
+    // value must be at least 1
+    value = value>0 ? value : 1;
+
+    // set value based on unit 
+    value = newPhases[index].delayUnit === 'days' ? value*24 : value;
+    newPhases[index].delay = value;
+
+    this.props.updatePhaseStarts(newPhases);
+    // logic for updating time stamps should be in parent - only change delays here
+  }
+
+  updateBatchSize(value, index) {
+    // When phase's batch size changes, adjust other phases' batch sizes starting with 'remainder' first.
+    let newPhases = this.props.phases;
+
+    if (value < 1) {
+      value = 1;
+    } else if (value >100) {
+      value = 100;
+    }
+
+    newPhases[index].batch_size = value;
+    const remainder = this.getRemainder(newPhases);
+    // if new remainder will be 0 or negative remove last phase
+    if (remainder < 1) {
+      newPhases.pop();
+      newPhases[newPhases.length-1].batch_size = this.getRemainder(newPhases);
```

Comment:
since the last phase not always has a `batch_size` this calculation seems to go wrong when the last phase has no batch_size defined and previous phases have too high percentages (the second to last phase gets a 0 or negative `batch_size`) and the phases "collapse" when calling `this.props.deploymentSettings(...)` so the approach selection is again shown
here it might be necessary to check the entered value and set it to the previous remainder if the new value is greater than the previous remainder